src/storage/db_storage.cpp

```diff
@@ -338,14 +338,6 @@ void DbStorage::savePeriodData(const SyncBlock& sync_block, Batch& write_batch)
     // Remove dag blocks
     remove(write_batch, Columns::dag_blocks, toSlice(block.getHash()));
   }
-  uint64_t position = 0;
```

Comment:
this should not be moved, because this approach is more error prone

Reply:
> I do not like it, because if we are using mutex, we could have trx status as in memory container and not use DB for that. The only think that should be saved in DB is final position inside the block, where we know, that was finalized.

I do not see how this could be done without a mutex. I mean maybe the mutex is not in the right class, maybe it should be implemented in DbStorage but if we have threads running at the same time that in parallel want to update the status of transaction based on the current status, I do not see a different way to do this